frontend/src/app/features/work-packages/components/wp-query-select/wp-query-select-dropdown.component.ts

```diff
@@ -94,444 +60,117 @@ export class WorkPackageQuerySelectDropdownComponent extends UntilDestroyedMixin
     no_results: this.I18n.t('js.work_packages.query.text_no_results'),
   };
 
-  private unregisterTransitionListener:Function;
-
-  private projectIdentifier:string|null;
-
-  private hiddenCategories:any = [];
-
-  private reportsBodySelector = '.controller-work_packages\\/reports';
-
-  private queryResultsContainer:JQuery;
+  private queryCategories:IOpSidemenuItem[] = [];
 
-  private buttonArrowLeft:JQuery;
-
-  private searchInput:IQueryAutocompleteJQuery;
+  private searchText = '';
 
   private initialized = false;
 
-  constructor(readonly ref:ChangeDetectorRef,
-    readonly element:ElementRef,
+  constructor(
     readonly apiV3Service:APIV3Service,
-    readonly $state:StateService,
-    readonly $transitions:TransitionService,
     readonly I18n:I18nService,
     readonly states:States,
     readonly CurrentProject:CurrentProjectService,
-    readonly loadingIndicator:LoadingIndicatorService,
-    readonly pathHelper:PathHelperService,
     readonly wpStaticQueries:WorkPackageStaticQueriesService,
     readonly mainMenuService:MainMenuNavigationService,
-    readonly toggleService:MainMenuToggleService,
-    readonly cdRef:ChangeDetectorRef) {
+    readonly cdRef:ChangeDetectorRef,
+  ) {
     super();
   }
 
-  public ngOnInit() {
-    this.queryResultsContainer = jQuery(this._queryResultsContainerElement.nativeElement);
-    this.projectIdentifier = this.element.nativeElement.getAttribute('data-project-identifier');
+  public get queries():IOpSidemenuItem[] {
+    return this.queryCategories
+      .map((category) => {
+        if (this.matchesSearchText(category.title)) {
+          return category;
+        }
+
+        const filteredChildren = category.children?.filter((query) => this.matchesSearchText(query.title));
+        return { title: category.title, children: filteredChildren };
+      })
+      .filter((category) => category.children && category.children.length > 0);
+  }
+
+  public set search(input:string) {
+    this.searchText = input;
+  }
 
+  ngOnInit():void {
     // When activating the work packages submenu,
     // either initially or through click on the toggle, load the results
     this.mainMenuService
       .onActivate('work_packages', 'work_packages_query_select')
       .subscribe(() => this.initializeAutocomplete());
-
-    // Register click handler on results
-    this.addClickHandler();
-    this.cdRef.detach();
-  }
-
-  ngOnDestroy() {
-    super.ngOnDestroy();
-    this.unregisterTransitionListener();
   }
 
-  private initializeAutocomplete() {
+  private initializeAutocomplete():void {
     if (this.initialized) {
       return;
     }
 
-    this.searchInput = jQuery(this._wpQueryMenuSearchInput.nativeElement) as any;
-    this.buttonArrowLeft = jQuery('.main-menu--arrow-left-to-project', jQuery('#main-menu-work-packages-wrapper').parent()) as any;
     this.initialized = true;
-    this.buttonArrowLeft.focus();
```

Comment:
This was done for accessibility reasons so that blind users know where there are. I think it makes sense to re-add this and not loose the functionality.

Reply:
readded without jquery